R/prune.R

```diff
@@ -0,0 +1,158 @@
+##' @importFrom utils capture.output
+
+extract_prune <- function(e, path, con) {
+
+  # The prune.csv file indicates...
+  # modelling_group, disease, scenario, and touchstones
+
+  # these can be '*' for wildcard,
+  # or a semi-colon separated list of options.
+
+  # No CSV provided
+  if (is.null(e$prune_csv)) {
+    return(NULL)
+  }
+
+  # Empty CSV provided
+  if (nrow(e$prune_csv) == 0) {
+    return(NULL)
+  }
+
+  # Check all necessary things exist, and
+  # return vector of things.
+
+  thing_exists <- function(entries, table) {
+    entries <- unique(unlist(strsplit(entries, ";")))
+    if ("*" %in% entries) {
+      return(NULL)
+    }
+    db_entries <- DBI::dbGetQuery(con, sprintf("
+      SELECT * FROM %s WHERE id IN ('%s')", table,
+        paste(entries, collapse="','")))$id
+    if (any(!entries %in% db_entries)) {
+      entries <- paste(entries[!entries %in% db_entries],
+                       collapse = ", ")
+      stop(sprintf("%s not found: %s\n", table, entries))
+    }
+    db_entries
+  }
+
+  modelling_group <- thing_exists(e$prune_csv$modelling_group, "modelling_group")
```

Comment:
Very minor but you could put in some handling for empty cell in csv, which I think it will end up as an NA in `e$prune_csv` which throws a bit of an obscure error in `strsplit`.

Reply:
I'm already stopping if `nrow(e$prune_csv) ==0` - have added a column check as well, which should prevent this from being possible.